docs/tutorials/hetzner.md

```diff
@@ -0,0 +1,191 @@
+# Setting up ExternalDNS for Services on Hetzner DNS
+
+This tutorial describes how to setup ExternalDNS for usage within a Kubernetes cluster using Hetzner DNS.
+
+Make sure to use **>=0.7.2** version of ExternalDNS for this tutorial.
+
+## Creating a Hetzner DNS zone
+
+If you want to learn about how to use Hetzner's DNS service read the following tutorial series:
+
+[An Introduction to Managing DNS](https://wiki.hetzner.de/index.php/DNS_Overview), and [Add a new DNS zone](https://wiki.hetzner.de/index.php/Getting_started).
+
+Create a new DNS zone where you want to create your records in. Let's use `example.com` as an example here.
+
+## Creating Hetzner Credentials
+
+Generate a new personal token by going to [the API settings](https://dns.hetzner.com/settings/api-token) or follow [Generating an API access token](https://wiki.hetzner.de/index.php/API_access_token) if you need more information. Give the token a name and choose read and write access. The token needs to be passed to ExternalDNS so make a note of it for later use.
+
+The environment variable `HETZNER_TOKEN` will be needed to run ExternalDNS with Hetzner.
+
+## Deploy ExternalDNS
+
+Connect your `kubectl` client to the cluster you want to test ExternalDNS with.
+Then apply one of the following manifests file to deploy ExternalDNS.
+
+### Manifest (for clusters without RBAC enabled)
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: external-dns
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:latest
+        args:
+        - --source=service # ingress is also possible
+        - --domain-filter=example.com # (optional) limit to only example.com domains; change to match the zone created above.
+        - --provider=hetzner
+        env:
+        - name: HETZNER_TOKEN
+          value: "YOUR_HETZNER_DNS_API_KEY"
+```
+
+### Manifest (for clusters with RBAC enabled)
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: external-dns
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: external-dns
+rules:
+- apiGroups: [""]
+  resources: ["services","endpoints","pods"]
+  verbs: ["get","watch","list"]
+- apiGroups: ["extensions"] 
+  resources: ["ingresses"] 
+  verbs: ["get","watch","list"]
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["list"]
```

Comment:
watch is missing

Reply:
missing watch on nodes? are you sure? manifest same as others.

Reply:
yes I'm pretty sure, https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/aws.md.
we're using the SharedInformer for node objects: https://github.com/kubernetes-sigs/external-dns/blob/master/source/node.go#L66.

Reply:
you right, I added watch